### PR TITLE
Upgrade go version from 1.22 to 1.23

### PR DESCRIPTION
**Pull request by fabiovincenzi**
Fixes https://github.com/mlflow/mlflow-go/issues/65
- Upgrade go version from 1.22 to 1.23
- Run `go mod tidy`
- Upgrade `golangci-lint`, `mockery` and `proton-gen-go`
- regenerate mocks and protos

### CODE REVIEW

**Comment by nojaf** [comment]
Please also update golangci-lint: https://github.com/golangci/golangci-lint/releases/tag/v1.61.0
and mockery: https://github.com/vektra/mockery/releases/tag/v2.46.3
Perhaps (https://pkg.go.dev/google.golang.org/protobuf/cmd/protoc-gen-go, do in separate commit, see what CI says, if negative don't go for it)

**Comment by nojaf** [comment]
@fabiovincenzi there were a couple of //nolint:ireturn comments that were removed.
You need to restore these. The clue is that the linter should now not mark these as false positives anymore.

File: pkg/tracking/store/sql/experiments_search_builder.go [hunk 1]
```diff
@@ -258,6 +258,7 @@ func applyExperimentsFilter(database, query *gorm.DB, filter string) (*gorm.DB,
 	return query, nil
 }
 
+//nolint:gosec // disable G115
```
Comment (nojaf) [inline pkg/tracking/store/sql/experiments_search_builder.go:261]: Could you create a follow-up issue for these violations?
(Did you try and fix them?)

**Review by nojaf — APPROVED** [review]
Thanks! Looks good, would like @dsuhinin to skim through this as well.

File: pkg/tracking/store/sql/experiments_search_builder.go (hunk 1 above)
Comment (fabiovincenzi) [reply pkg/tracking/store/sql/experiments_search_builder.go:261]: Created, I checked around for those violations and opened this issue https://github.com/mlflow/mlflow-go/issues/73

**Pull request merged**